api/client/build.go

```diff
@@ -223,19 +234,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		Remove:         *rm,
 		ForceRemove:    *forceRm,
 		PullParent:     *pull,
-		Isolation:      *isolation,
 		CPUSetCPUs:     *flCPUSetCpus,
 		CPUSetMems:     *flCPUSetMems,
 		CPUShares:      *flCPUShares,
 		CPUQuota:       *flCPUQuota,
 		CPUPeriod:      *flCPUPeriod,
 		CgroupParent:   *flCgroupParent,
-		ShmSize:        *flShmSize,
 		Dockerfile:     relDockerfile,
+		ShmSize:        shmSize,
 		Ulimits:        flUlimits.GetList(),
-		BuildArgs:      flBuildArg.GetAll(),
 		AuthConfigs:    cli.configFile.AuthConfigs,
 	}
+	options.IsolationLevel = container.IsolationLevel(*isolation)
```

Comment:
Already explained why I chose IsolationLevel to be embedded in https://github.com/docker/docker/pull/18977#discussion_r48573774. If you feel strongly against it, I can change.

BuildArgs is different. Earlier, it was an array of strings in dockerfile.Config and a hashmap of strings ->strings in ImageBuilder. The latter is elegant.